### DISCUSSION TITLE
how to do a where expression function that references a joined table?

### DISCUSSION THREAD

**Discussion by matthewkooshad** [simolus3/drift#1273] [Q&A] [answered]
i noticed in my where expression, the param to the function is referring to the table i have the select on, but i have inner joined a 2nd table that i need write some where conditions on. can you give an example on this? 
i see join here but not using where with it for what i'm trying to do: https://moor.simonbinder.eu/docs/advanced-features/joins/

**Comment by simolus3** [comment, marked as answer]
I've added documentation for this [here](https://60ce0f44fef656bae6e1169a--moor.netlify.app/docs/advanced-features/joins/#order-by-and-where-on-joins).
Basically, you can call `where` after `.join()`. It works similar to regular where clauses, except that you have to specify the table directly:

```dart
// no join:
select(table)..where((row) => row.id.equals(3))

// join
final query = select(table).join([...]);
query.where(table.id.equals(3));
``` 

If you need help with a specific query let me know!

**Reply by matthewkooshad** [reply]
thanks @simolus3 that helped me, but i think need to use ..where in another case after .join. i've opened up a new discussion, on this https://github.com/simolus3/moor/discussions/1274, since i think your answer here answers this q that i asked, trying to make my learning to be helpful search results for others.
